src/Runner.Worker/ActionManager.cs

```diff
@@ -504,27 +502,83 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 executionContext.Output($"Download action repository '{repositoryReference.Name}@{repositoryReference.Ref}'");
             }
 
+            var configurationStore = HostContext.GetService<IConfigurationStore>();
+            var isHostedServer = configurationStore.GetSettings().IsHostedServer;
+            if (isHostedServer)
+            {
+                string apiUrl = GetApiUrl(executionContext, forceDotCom: true);
+                string archiveLink = BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref);
+                Trace.Info($"Download archive '{archiveLink}' to '{destDirectory}'.");
+                await DownloadRepositoryActionAsync(executionContext, archiveLink, destDirectory);
```

Comment:
`return;` after the `await DownloadRepositoryActionAsync()`

Reply:
Great catch